fraqbot/Local/randomemoji.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import os
+import random
+import sys
+
+from Legobot.Connectors.Slack import Slack
+from Legobot.Lego import Lego
+
+
+logger = logging.getLogger(__name__)
+LOCAL_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)))
+
+if LOCAL_DIR not in sys.path:
+    sys.path.append(LOCAL_DIR)
+
+
+from helpers import utils  # noqa: E402
+
+
+class RandomEmoji(Lego):
+    def __init__(self, baseplate, lock, *args, **kwargs):
+        super().__init__(baseplate, lock, acl=kwargs.get('acl'))
+        self._set_client()
+        self.max_how_many = 20
+        self.min_how_many = 1
+        self.default_how_many = 5
+
+    def listening_for(self, message):
+        text = message.get('text')
+
+        return isinstance(text, str) and text.startswith('!emoji')
+
+    def _set_client(self):
+        self.client = None
+        children = self.baseplate._actor.children
+
+        if children:
+            slack = [a._actor for a in children if isinstance(a._actor, Slack)]
+            if slack:
+                self.client = slack[0].botThread.slack_client
+
+    def _fetch_slack_emojis(self):
+        return utils.call_slack_api(
+                self.client,
+                'emoji.list',
+                False,
+                'emoji'
+            )
+
+    def _get_emoji(self, how_many):
+        how_many_limited = max(
+            self.min_how_many,
+            min(self.max_how_many, how_many)
+        )
+
+        emoji_response = self._fetch_slack_emojis()
+
+        return (':'
+                + ': :'.join(random.choices(
+                        list(emoji_response.keys()), k=how_many_limited
+                    )).strip()
+                + ':')
+
+    def handle(self, message):
+        logger.debug(
+            'Handling Random Emoji request: {}'.format(message['text'])
+        )
+        text_provided = message['text'][6:].strip()
+
+        opts = self.build_reply_opts(message)
+        if (not text_provided.isdigit()):
```

Comment:
superfluous parens.
This if statement is triggered when you don't pass any second value (i.e. just `!emoji`).
results:
```
15:51
fraqbotAPP '' is not a valid integer.
15:51
:raceconditionparrot: :incredible: :link-success: :jeff_corcoran: :chandler_dance:
```
If this reply is triggered, the Lego still continues processing, so it throws an exception when it tries to set `how_many`
Processing should probably stop here.

Reply:
Oops good callouts